libcontainer/container_linux.go

```diff
@@ -2148,11 +2146,16 @@ func (c *linuxContainer) bootstrapData(cloneFlags uintptr, nsMaps map[configs.Na
 	if !joinExistingUser {
 		// write uid mappings
 		if len(c.config.UidMappings) > 0 {
-			if c.config.RootlessEUID && c.newuidmapPath != "" {
-				r.AddData(&Bytemsg{
-					Type:  UidmapPathAttr,
-					Value: []byte(c.newuidmapPath),
-				})
+			if c.config.RootlessEUID {
+				// We resolve the paths for new{u,g}idmap from
+				// the context of runc to avoid doing a path
+				// lookup in the nsexec context.
+				if path, err := exec.LookPath("newuidmap"); err == nil {
```

Comment:
Not really related to this PR, but it _may_ makes sense to bail out right here.

We are rootless, and have a UID mapping, so in the absence of newuidmap/newgidmap nsexec.c is doomed to fail anyway. Or could it be the case when writing to `/proc/<pid>/uid_map might succeed? I don't think so but I might be wrong.

WDYT @AkihiroSuda

Reply:
Should we use `golang.org/x/sys/execabs` for these? (https://github.com/golang/sys/tree/master/execabs)

See https://github.com/docker/cli/pull/2950 -  `golang.org/x/sys/execabs` is an updated version of `os/exec` that fixes the security issue that was found in Go 1.15.7; https://blog.golang.org/path-security

Reply:
For some reason I was not able to reply to these comments from the main PR page (https://github.com/opencontainers/runc/pull/3354), so I replied "globally".